backend/src/services/collectionService.ts

```diff
@@ -398,142 +396,133 @@ export class CollectionService extends LoggerBase {
   }
 
   async findRepositoriesForSegment(segmentId: string) {
-    const qx = SequelizeRepository.getQueryExecutor(this.options)
-    const integrations = await fetchIntegrationsForSegment(qx, segmentId)
-
-    // Initialize result with platform arrays
-    const result: Record<string, Array<{ url: string; label: string }>> = {
-      git: [],
-      github: [],
-      gitlab: [],
-      gerrit: [],
-    }
-
-    const addToResult = (platform: PlatformType, fullUrl: string, label: string) => {
-      const platformKey = platform.toLowerCase()
-      if (!result[platformKey].some((item) => item.url === fullUrl)) {
-        result[platformKey].push({ url: fullUrl, label })
+    return SequelizeRepository.withTx(this.options, async (tx) => {
+      const qx = SequelizeRepository.getQueryExecutor(this.options, tx)
+      const integrations = await fetchIntegrationsForSegment(qx, segmentId)
+
+      // Initialize result with platform arrays
+      const result: Record<string, Array<{ url: string; label: string }>> = {
+        git: [],
+        github: [],
+        gitlab: [],
+        gerrit: [],
       }
-    }
 
-    for (const i of integrations) {
-      if (i.platform === PlatformType.GITHUB) {
-        for (const org of (i.settings as any).orgs) {
-          for (const repo of org.repos) {
-            const label = `${org.name}/${repo.name}`
-            const fullUrl = `https://github.com/${label}`
-            addToResult(i.platform, fullUrl, label)
-          }
+      const addToResult = (platform: PlatformType, fullUrl: string, label: string) => {
+        const platformKey = platform.toLowerCase()
+        if (!result[platformKey].some((item) => item.url === fullUrl)) {
+          result[platformKey].push({ url: fullUrl, label })
         }
       }
 
-      if (i.platform === PlatformType.GITHUB_NANGO) {
-        for (const org of (i.settings as any).orgs) {
-          for (const repo of org.repos) {
-            const label = `${org.name}/${repo.name}`
-            const fullUrl = `https://github.com/${label}`
-            addToResult(PlatformType.GITHUB, fullUrl, label)
+      // Add mapped repositories to GitHub platform
+      const segmentRepository = new SegmentRepository(this.options)
+      const mappedRepos = await segmentRepository.getMappedRepos(segmentId)
```

Comment:
All the logic on this method is ok. But now that I look into `getMappedRepos`, I think we should simplify it because it won't return exactly what we want.

The "getMappedRepos" gets repositories that are assigned to a segment but whose associated integration belongs to a different segment (or no integration at all).

Meaning that it will work well for projects that are repos mapped from other integrations, but it won't work well for projects that have repos within its own integration.

I think if we change the query on `getMappedRepos`to:
```
select
         r.url as url
       from
        "githubRepos" r
       where r."segmentId" = :segmentId
       and r."tenantId" = :tenantId
       order by r.url
```
It should work